core/src/main/java/com/netflix/conductor/core/events/queue/EventPollSchedulerProvider.java

```diff
@@ -37,11 +37,14 @@ public EventPollSchedulerProvider(Configuration configuration) {
 
     @Override
     public Scheduler get() {
-        ThreadFactory threadFactory = new ThreadFactoryBuilder()
-                .setNameFormat("event-queue-poll-scheduler-thread-%d")
-                .build();
-        Executor executorService = Executors.newFixedThreadPool(configuration.getEventSchedulerPollThreadCount(), threadFactory);
-
-        return Schedulers.from(executorService);
+        if (configuration.getEventSchedulerPollThreadCount() != null) {
+            ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                    .setNameFormat("event-queue-poll-scheduler-thread-%d")
+                    .build();
+            Executor executorService = Executors.newFixedThreadPool(configuration.getEventSchedulerPollThreadCount(), threadFactory);
+            return Schedulers.from(executorService);
+        } else {
+            return Schedulers.computation();
```

Comment:
@mdepak Based on the documentation here, http://reactivex.io/RxJava/javadoc/rx/schedulers/Schedulers.html#computation--

> Do not perform IO-bound work on this scheduler. Use io() instead.

and the fact that IO needs to be performed as seen [here](https://github.com/Netflix/conductor/blob/master/core/src/main/java/com/netflix/conductor/core/events/queue/dyno/DynoObservableQueue.java#L129-L137)

Does it make sense to use `io()` instead here?

Reply:
@aravindanr  `io()` is a good candidate but the potential problem with this is it can grow the thread pool exponentially in case the IO wait is taking too long because of a problem with the Queue system and could cause OOM. RxJava 2.x doc has a more detailed explanation regarding the same http://reactivex.io/RxJava/2.x/javadoc/io/reactivex/schedulers/Schedulers.html#io--

The objective of this fix is to ensure that the event processing functionality works as expected by default without tuning up the thread pool size. Meanwhile, I plan to debug why [Executor scheduler](http://reactivex.io/RxJava/javadoc/rx/schedulers/Schedulers.html#from-java.util.concurrent.Executor-) is not working as expected, and its behavior in the later versions of RxJava. Please let me know your feedback.